File: custom-validation/src/main/java/yun/custom/validation/member/Member.java
package yun.custom.validation.member;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.validation.constraints.Email;
import lombok.Builder;
import lombok.Getter;

@Entity
@Table(name = "member")
@Getter
public class Member {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Email
    @Column(name = "email", nullable = false, updatable = false, unique = true)
    private String email;

    @Builder
    public Member(String email) {
        this.email = email;
    }
}

File: spring-test/src/main/kotlin/com/spring/test/member/Member.kt
package com.spring.test.member

import com.spring.test.EntityAuditing
import javax.persistence.Column
import javax.persistence.Entity

@Entity
class Member(
    @Column(name = "name", nullable = false)
    var name: String,

    @Column(name = "email", nullable = false, updatable = true)
    var email: String
) : EntityAuditing() {

}